src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -345,8 +353,15 @@ private void OpenIndexes()
                     if (int.TryParse(indexDirectory.Name, out indexId) == false)
                         continue;
 
-                    var index = Index.Open(indexId, _documentDatabase);
-                    _indexes.Add(index);
+                    try
+                    {
+                        var index = Index.Open(indexId, _documentDatabase);
+                        _indexes.Add(index);
+                    }
+                    catch (Exception e)
+                    {
+                        Log.ErrorException($"Could not open index with id {indexId}", e);
```

Comment:
Yes. Such behavior is definitely not correct. I will change things to create an errored, in-memory impl for that case. Right now we don't have any alerting mechanism - will put TODO to keep it in mind.